### PR TITLE
Check for new updates of CLI itself

### PR DESCRIPTION
**Pull request by holisticode**

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -65,6 +68,9 @@ in with avalanche subnet create myNewSubnet.`,
 
 	// add transaction command
 	rootCmd.AddCommand(transactioncmd.NewCmd(app))
+
+	// add update command
+	rootCmd.AddCommand(updatecmd.NewCmd(app, Version))
```
Comment (dasconnor) [inline cmd/root.go:73]: I like that you added a command to manually initiate

File: cmd/root.go [hunk 2]
```diff
@@ -99,6 +105,31 @@ func createApp(cmd *cobra.Command, _ []string) error {
...
+	isUserCalled := false
+	if err := updatecmd.Update(cmd, isUserCalled); err != nil {
+		if err == updatecmd.ErrUserAbortedInstallation {
+			return nil
+		}
+		if err == updatecmd.ErrNoVersion {
+			ux.Logger.PrintToUser(
+				"Attempted to check if a new version is available, but couldn't find the currently running version information")
+			ux.Logger.PrintToUser(
+				"This can be ignored, but make sure to follow official instructions, or automatic updates won't be available for you")
```
Comment (dasconnor) [inline cmd/root.go:127]: Remove the `This can be ignored` part

File: cmd/updatecmd/update.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+	app *application.Avalanche
+	yes bool
+)
+
+func NewCmd(injectedApp *application.Avalanche, version string) *cobra.Command {
+	app = injectedApp
+	cmd := &cobra.Command{
+		Use:          "update",
+		Short:        "Check for latest updates of this same tool",
```
Comment (dasconnor) [inline cmd/updatecmd/update.go:33]: `of Avalanche-CLI`

File: cmd/updatecmd/update.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+		Long:         `Check if an update is available, and prompt the user to install it`,
+		RunE:         runUpdate,
+		Args:         cobra.ExactArgs(0),
+		SilenceUsage: true,
+		Version:      version,
+	}
+
+	cmd.Flags().BoolVarP(&yes, "assume-yes", "y", false, "Assume yes for installation")
```
Comment (dasconnor) [inline cmd/updatecmd/update.go:41]: I think I know what you mean, but this isn't clear. We should rename the flag.

Comment (dasconnor) [reply cmd/updatecmd/update.go:41]: maybe `--confirm`?

File: cmd/root.go [hunk 5]
```diff
@@ -99,6 +105,31 @@ func createApp(cmd *cobra.Command, _ []string) error {
 		return err
 	}
 
+	if !skipCheck {
```
Comment (dasconnor) [inline cmd/root.go:108]: We should only check this like once a day

**Review by dasconnor — COMMENTED** [review]
Few small suggestions

File: cmd/root.go (hunk 5 above)
Comment (holisticode) [reply cmd/root.go:108]: Provided a proposal, please review

**Review by dasconnor — APPROVED** [review]

**Pull request merged**